### PR TITLE
Set up redirects based on 404 data & filter sitemap

### PR DESCRIPTION
**Pull request by delucis**
#### What kind of changes does this PR include?

- [x] Changes to the docs site code

#### Description

- Add a Netlify redirect for an old `renderer-reference` URL

- Add a route with a `<meta>` refresh redirect for `/[lang]/tutorial`

    We may one day _have_ an actual `/tutorial` landing page potentially, so using a `.astro` route with `<meta>` seemed safer than configuring a Netlify redirect we’d be more likely to forget to remove.

#### Testing

These URLs were 404s but should now work:

- https://deploy-preview-1976--astro-docs-2.netlify.app/reference/renderer-reference/
- https://deploy-preview-1976--astro-docs-2.netlify.app/en/tutorial/

### CODE REVIEW

File: src/pages/[lang]/tutorial.astro [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+import languages from '../../i18n/languages';
+
+export const getStaticPaths = () => Object.keys(languages).map((lang) => ({ params: { lang } }));
+
+const { lang } = Astro.params;
+---
+
+<meta http-equiv="refresh" content={`0;url=/${lang}/tutorial/0-introduction/`} />
```
Comment (tony-sull) [inline src/pages/[lang]/tutorial.astro:9]: Will this show up in the sitemap?

Comment (delucis) [reply src/pages/[lang]/tutorial.astro:9]: Excellent question. The answer is: yes!

https://deploy-preview-1976--astro-docs-2.netlify.app/sitemap-0.xml

We can filter sitemap URLs I believe. Let me look at that — we should probably exclude our other similar redirects too.

**Comment by sarah11918** [comment]
No objections here, once you're happy @delucis !

Just putting it on the radar that since we've discussed adding additional content in the "Tutorials" section, maybe it's time for @Jutanium to start discussing/sketching out a generic tutorial landing page anyway? I think this could fit in next sprint, actually!

File: src/pages/[lang]/tutorial.astro (hunk 1 above)
Comment (delucis) [reply src/pages/[lang]/tutorial.astro:9]: OK, question back time: does it matter if it is in sitemap?

Comment (tony-sull) [reply src/pages/[lang]/tutorial.astro:9]: Its usually best to keep redirects out of the sitemap, but that can be tackled in a separate PR :+1:

Its really not a huge deal in general, I think it's pretty common for redirects to linger in a sitemap for a while (or forever on some sites)

**Review by tony-sull — APPROVED** [review]
lgtm!

File: src/pages/[lang]/tutorial.astro (hunk 1 above)
Comment (delucis) [reply src/pages/[lang]/tutorial.astro:9]: OK — pushed a fix for the low-hanging fruit here. Our `[lang]/something.astro` redirects like the one added in this PR are easy to filter out. Other dynamic redirects in the root of `src/pages/` will be trickier and can be tackled later I guess!

File: astro.config.ts [hunk 2]
```diff
@@ -47,6 +53,7 @@ export default defineConfig({
 	integrations: [
 		preact({ compat: true }),
 		sitemap({
+			filter: (page) => sitemapBlocklist.has(new URL(page).pathname),
```
Comment (delucis) [inline astro.config.ts:56]: Would help if I got the logic the right way round 😅 

```suggestion
			filter: (page) => !sitemapBlocklist.has(new URL(page).pathname),
```

File: src/pages/[lang]/tutorial.astro (hunk 1 above)
Comment (delucis) [reply src/pages/[lang]/tutorial.astro:9]: Engaged morning brain and implemented a much better filtering set-up so now the sitemap shouldn’t include any of our redirect routes!

**Review by natemoo-re — APPROVED** [review]

File: integrations/sitemap.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+/** Match a pathname starting with “lighthouse” or one of our language tags. */
+const ValidRouteRE = new RegExp(`^/(lighthouse|${langTags.join('|')})/`);
+
+/** Test a pathname is not in our blocklist and starts with a valid prefix. */
+const isValidPath = (path: string) => !blocklist.has(path) && ValidRouteRE.test(path);
+
+/** Get a preconfigured instance of the `@astrojs/sitemap` integration. */
+export function sitemap(): AstroIntegration {
+	return AstroSitemap({
+		filter: (page) => isValidPath(new URL(page).pathname),
```
Comment (tony-sull) [inline integrations/sitemap.ts:24]: Nice!

**Review by tony-sull — APPROVED** [review]

**Pull request merged**